pypeit/core/wavecal/autoid.py

```diff
@@ -0,0 +1,799 @@
+""" Module for finding patterns in arc line spectra
+"""
+from __future__ import (print_function, absolute_import, division, unicode_literals)
+
+from scipy.ndimage.filters import gaussian_filter
+import numpy as np
+import pdb
+
+from pypeit.core.wavecal import waveio
+from pypeit.core.wavecal import patterns
+from pypeit.core.wavecal import fitting
+from pypeit.core.wavecal import utils
+from pypeit.core.wavecal import qa
+
+from pypeit import msgs
+
+
+def basic(spec, lines, wv_cen, disp, siglev=20., min_ampl=300.,
+          swv_uncertainty=350., pix_tol=2, plot_fil=None, min_match=5,
+          **kwargs):
+    """ Basic holy grail algorithm
+
+    Parameters
+    ----------
+    spec : spectrum
+    lines : list
+      List of arc lamps on
+    wv_cen : float
+      Guess at central wavelength
+    disp : float
+      Dispersion A/pix
+    siglev
```

Comment:
can you add to this doc string?  
I guess I'm too blame for it. ;)

Reply:
done